### ISSUE TITLE
[bug]: is_smtp_configured is not working as supposed

### ISSUE THREAD

**Issue by rouja** [makeplane/plane#3734]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

If we use a smtp gateway without user and password, is_smtp_configured does not work : 

https://github.com/makeplane/plane/blob/3efb7fc0701e7dd0b2e2885fd063f59ca38328c3/apiserver/plane/app/views/config.py#L127C15-L127C33

### Steps to reproduce

1) Configure email without user and password 

![2024-02-21_12-25](https://github.com/makeplane/plane/assets/22145127/48229b71-3f88-4aa3-bd4f-575af455f65f)

2) You are not able to connect with a magic link

I don't know how to fix it. Do you have a recommandation ? I probably can to implement the solution.

### Browser

Google Chrome

### Version

Self-hosted

**Comment by srinivaspendem** [comment]
Hello @rouja we currently need email and password to make the smtp integration work. I apologise if this is not getting conveyed in the form, we will update it to reflect those as required fields.

**Comment by rouja** [comment]
I made [this PR](https://github.com/makeplane/plane/pull/3732) to avoid this behavior because we do not need always user and password to send email. It depends of your infrastructure configuration.

**Comment by rouja** [comment]
@pablohashescobar Do you have an idea about the best way to fix this point ?

**Comment by srinivaspendem** [comment]
@rouja i don't think we will be prioritising it currently and we don't have an immediate fix in mind.
